### PR TITLE
network: introduce ManageTemporaryAddress= setting in [DHCPv6PrefixDelegation] section

### LINKED ISSUE #18208: Systemd-networkd: allow `ManageTemporaryAddress` in `[DHCPv6PrefixDelegation]`
**Is your feature request related to a problem? Please describe.**

 > I'm always frustrated when I have to manually create a static address (in `[Address]`) when I already get a prefix from `[DHCPv6PrefixDelegation]`. 

**Describe the solution you'd like**

 > Being able to specify `ManageTemporaryAddress` in `[DHCPv6PrefixDelegation]` directly.
 > `Token` does not change its meaning: the assigned address is just added to the system with the `mngtmpaddr` flag.

**Describe alternatives you've considered**

> Using static addresses. All other alternatives I thought of are not neat (systemd unit file to scan `ip -6 a s` then construct and add new address with the right prefix).

### PR DESCRIPTION
**Pull request by yuwata**
Closes #18208.

### CODE REVIEW

File: src/network/networkd-dhcp6.c [hunk 1]
```diff
@@ -416,16 +417,20 @@ static int dhcp6_set_pd_address(Link *link,
...
 }
 
-static int dhcp6_pd_assign_prefix(Link *link, const union in_addr_union *prefix, const union in_addr_union *pd_prefix,
-                                  uint8_t prefix_len, uint32_t lifetime_preferred, uint32_t lifetime_valid) {
+static int dhcp6_pd_assign_prefix(
+                Link *link,
+                const union in_addr_union *prefix,
+                const union in_addr_union *pd_prefix,
+                uint32_t lifetime_preferred,
+                uint32_t lifetime_valid) {
```
Comment (bluca) [inline src/network/networkd-dhcp6.c:425]: style nit: empty line after { when breaking arguments in multiple lines

File: man/systemd.network.xml [hunk 2]
```diff
@@ -2056,6 +2056,13 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
           addresses. Defaults to unset.</para>
         </listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>ManageTemporaryAddress=</varname></term>
+        <listitem>
+          <para>As in the [Address] section. But, defaults to true.</para>
```
Comment (bluca) [inline man/systemd.network.xml]: "...section, but defaults..." sounds better to me

**Comment by yuwata** [comment]
@bluca Thank you for the comments. Updated.

**Review by bluca — APPROVED** [review]
lgtm

**Comment by poettering** [comment]
lgtm, too

**Pull request merged**